lib/connectors/crawler/scheduler.rb

```diff
@@ -22,11 +22,48 @@ def connector_settings
         []
       end
 
+      def when_triggered
+        loop do
+          connector_settings.each do |cs|
+            # crawler only supports :sync
+            if sync_triggered?(cs)
+              yield cs, :sync, nil
+              next
+            end
+
+            custom_schedule_triggered, schedule_key = custom_schedule_triggered?(cs)
+            if custom_schedule_triggered
+              yield cs, :sync, schedule_key
+            end
+          end
+        rescue *Utility::AUTHORIZATION_ERRORS => e
+          log_authorization_error(e)
+        rescue StandardError => e
+          log_standard_error(e)
+        ensure
+          if @is_shutting_down
+            break
+          end
+          sleep_for_poll_interval
+        end
+      end
+
       private
 
       def connector_registered?(service_type)
         service_type == 'elastic-crawler'
       end
+
+      def custom_schedule_triggered?(cs)
+        cs.custom_scheduling_settings.each do |key, custom_scheduling|
```

Comment:
Could you please leave a comment that only the first found scheduling will be triggered?